avocado/core/test.py

```diff
@@ -144,7 +144,8 @@ class Test(unittest.TestCase):
     timeout = None
 
     def __init__(self, methodName='test', name=None, params=None,
-                 base_logdir=None, job=None, runner_queue=None):
+                 base_logdir=None, job=None, runner_queue=None,
+                 recovery_required=False, previous_run=None):
```

Comment:
Should the test have knowledge about the recovery? I think in this more like a replay functionality. The very test that was under execution when the interruption takes place should be either skipped or executed. I don' t see a reason to raise a `TestError`.